[NativeAOT] When attempting to AOT compile a benchmark `System.InvalidCastException: Specified cast is not valid` is thrown

### Description

It appears that with the current RC.1 build it is impossible to build AOT benchmarks with BenchmarkDotNet due to an exception that occurs within ILC.

### Reproduction Steps

1. `dotnet new console`
2. `dotnet add package BenchmarkDotNet` (0.14.0)
3. Define simple benchmark:
```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Jobs;
using BenchmarkDotNet.Running;

BenchmarkRunner.Run<Benchmark>();

[SimpleJob(RuntimeMoniker.NativeAot90)]
public class Benchmark {
    [Benchmark]
    public void DoNothing() { }
}
```
4. `dotnet run -c Release`

### Expected behavior

Benchmark successfully runs

### Actual behavior

An exception is thrown:
```
// Build Error: Standard output: 
  
 Standard error: 
 /usr/local/share/dotnet/sdk/9.0.100-rc.1.24414.7/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(326,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/Users/arseniy/Code/Local/ILScannerRepro/bin/Release/net9.0/7612d247-f6bf-43c3-b660-89158fea3a74/BenchmarkDotNet.Autogenerated.csproj]
  ILScannerRepro -> /Users/arseniy/Code/Local/ILScannerRepro/bin/Release/net9.0/7612d247-f6bf-43c3-b660-89158fea3a74/bin/Release/net9.0/osx-arm64/publish/ILScannerRepro.dll
  BenchmarkDotNet.Autogenerated -> /Users/arseniy/Code/Local/ILScannerRepro/bin/Release/net9.0/7612d247-f6bf-43c3-b660-89158fea3a74/bin/Release/net9.0/osx-arm64/publish/7612d247-f6bf-43c3-b660-89158fea3a74.dll
/Users/arseniy/.nuget/packages/microsoft.net.illink.tasks/9.0.0-rc.1.24410.5/build/Microsoft.NET.ILLink.targets(220,5): warning : Property 'TrimmerDefaultAction' is deprecated in .NET 7 and higher and will be ignored. Use TrimMode instead. [/Users/arseniy/Code/Local/ILScannerRepro/bin/Release/net9.0/7612d247-f6bf-43c3-b660-89158fea3a74/BenchmarkDotNet.Autogenerated.csproj]
  Generating native code
EXEC : error : Specified cast is not valid. [/Users/arseniy/Code/Local/ILScannerRepro/bin/Release/net9.0/7612d247-f6bf-43c3-b660-89158fea3a74/BenchmarkDotNet.Autogenerated.csproj]
  System.InvalidCastException: Specified cast is not valid.
     at ILCompiler.CompilerTypeSystemContext.EnsureLoadableTypeUncached(TypeDesc) + 0x428
     at ILCompiler.CompilerTypeSystemContext.EnsureLoadableType(TypeDesc) + 0x54
     at ILCompiler.LibraryRootProvider.CheckCanGenerateMethod(MethodDesc) + 0x64
     at ILCompiler.DependencyAnalysis.TypeMetadataNode.GetStaticDependencies(NodeFactory) + 0x430
     at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.GetStaticDependenciesImpl(DependencyNodeCore`1) + 0x4c
     at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.GetStaticDependencies(DependencyNodeCore`1) + 0xf4
     at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.ProcessMarkStack() + 0xe8
     at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.ComputeMarkedNodes() + 0x40
     at ILCompiler.ILScanner.ILCompiler.IILScanner.Scan() + 0x20
     at ILCompiler.Program.<Run>g__RunScanner|4_0(Program.<>c__DisplayClass4_0&) + 0x1e0
     at ILCompiler.Program.Run() + 0x2368
     at ILCompiler.ILCompilerRootCommand.<>c__DisplayClass240_0.<.ctor>b__0(ParseResult) + 0x308
/Users/arseniy/.nuget/packages/microsoft.dotnet.ilcompiler/9.0.0-rc.1.24410.5/build/Microsoft.NETCore.Native.targets(317,5): error MSB3073: The command ""/Users/arseniy/.nuget/packages/runtime.osx-arm64.microsoft.dotnet.ilcompiler/9.0.0-rc.1.24410.5/tools/ilc" @"/Users/arseniy/Code/Local/ILScannerRepro/bin/Release/net9.0/7612d247-f6bf-43c3-b660-89158fea3a74/obj/Release/net9.0/native/7612d247-f6bf-43c3-b660-89158fea3a74.ilc.rsp"" exited with code 1. [/Users/arseniy/Code/Local/ILScannerRepro/bin/Release/net9.0/7612d247-f6bf-43c3-b660-89158fea3a74/BenchmarkDotNet.Autogenerated.csproj]
```

### Regression?

Yes

### Known Workarounds

None that I'm aware of

### Configuration

```
.NET SDK:
 Version:           9.0.100-rc.1.24414.7
 Commit:            71a23de93f
 Workload version:  9.0.100-manifests.14db7d11
 MSBuild version:   17.12.0-preview-24407-03+6bc91d5e2

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  15.0
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/9.0.100-rc.1.24414.7/
 ```

### Other information

_No response_

--- a/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeMetadataNode.cs
+++ b/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/TypeMetadataNode.cs
@@ -77,9 +77,11 @@ namespace ILCompiler.DependencyAnalysis
                     {
                         try
                         {
-                            // Make sure we're not adding a method to the dependency graph that is going to
-                            // cause trouble down the line. This entire type would not actually load on CoreCLR anyway.
-                            LibraryRootProvider.CheckCanGenerateMethod(method);
+                            // Spot check by parsing signature.
+                            // Previously we had LibraryRootProvider.CheckCanGenerateMethod(method) here, but that one
+                            // expects fully instantiated types and methods. We operate on definitions here.
+                            // This is not as thorough as it could be. This option is unsupported anyway.
+                            _ = method.Signature;
                         }
                         catch (TypeSystemException)
                         {

